Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -0,0 +1,314 @@
+                           "," + boost::lexical_cast<std::string>(h + box) +
+                           "," + std::to_string(gridPts));
+  normAlg->setProperty("AlignedDim1",
+                       "[0,K,0]," + boost::lexical_cast<std::string>(k - box) +
+                           "," + boost::lexical_cast<std::string>(k + box) +
+                           "," + std::to_string(gridPts));
+  normAlg->setProperty("AlignedDim2",
+                       "[0,0,L]," + boost::lexical_cast<std::string>(l - box) +
+                           "," + boost::lexical_cast<std::string>(l + box) +
+                           "," + std::to_string(gridPts));
+  normAlg->setProperty("FluxWorkspace", flux);
+  normAlg->setProperty("SolidAngleWorkspace", sa);
+  normAlg->setProperty("OutputWorkspace", "mdout");
+  normAlg->setProperty("OutputNormalizationWorkspace", "mdnorm");
+  normAlg->executeAsChildAlg();
+  Workspace_sptr mdout = normAlg->getProperty("OutputWorkspace");
+  Workspace_sptr mdnorm = normAlg->getProperty("OutputNormalizationWorkspace");
+
+  IAlgorithm_sptr alg = createChildAlgorithm("DivideMD");
+  alg->setProperty("LHSWorkspace", mdout);
+  alg->setProperty("RHSWorkspace", mdnorm);
+  alg->setPropertyValue("OutputWorkspace", "out");
+  alg->execute();
+  IMDWorkspace_sptr out = alg->getProperty("OutputWorkspace");
+  return boost::dynamic_pointer_cast<MDHistoWorkspace>(out);
+}
+
+void IntegratePeaksMDHKL::integratePeak(const int neighborPts,
+                                        MDHistoWorkspace_sptr out,
+                                        double &intensity,
+                                        double &errorSquared) {
+  std::vector<int> gridPts;
+  const size_t dimensionality = out->getNumDims();
+  for (size_t i = 0; i < dimensionality; ++i) {
+    gridPts.push_back(static_cast<int>(out->getDimension(i)->getNBins()));
+  }
+
+  double *F = out->getSignalArray();
+  double Fmax = 0.;
+  double Fmin = std::numeric_limits<double>::max();
+  double sum = 0.0;
+  for (int i = 0; i < gridPts[0] * gridPts[1] * gridPts[2]; i++) {
+    if (std::isnormal(F[i])) {
+      sum += F[i];
+      Fmin = std::min(Fmin, F[i]);
+      Fmax = std::max(Fmax, F[i]);
+    }
+  }
+  double *SqError = out->getErrorSquaredArray();
+
+  double minIntensity = Fmin + 0.01 * (Fmax - Fmin);
+  int measuredPoints = 0;
+  int peakPoints = 0;
+  double peakSum = 0.0;
+  double measuredSum = 0.0;
+  double errSqSum = 0.0;
+  double measuredErrSqSum = 0.0;
+  for (int Hindex = 0; Hindex < gridPts[0]; Hindex++) {
+    for (int Kindex = 0; Kindex < gridPts[1]; Kindex++) {
+      for (int Lindex = 0; Lindex < gridPts[2]; Lindex++) {
+        int iHKL = Hindex + gridPts[0] * (Kindex + gridPts[1] * Lindex);
+        if (std::isfinite(F[iHKL])) {
+          measuredPoints = measuredPoints + 1;
+          measuredSum = measuredSum + F[iHKL];
+          measuredErrSqSum = measuredErrSqSum + SqError[iHKL];
```

Comment:
If in BinMD all three grids were 40,000 instead of 500, you should get the same answer in peaks_histo and peaks_bin. I suspect some of the background is included in intensity for coarse grid.